File: app/src/main/java/com/example/techfix/ui/ServiceAdapter.java
package com.example.techfix.ui;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.techfix.R;
import com.example.techfix.model.RepairService;

import java.util.List;
import java.util.Locale;

public class ServiceAdapter extends RecyclerView.Adapter<ServiceAdapter.ServiceViewHolder> {

    public interface OnServiceClickListener {
        void onServiceClick(RepairService service);
    }

    private final List<RepairService> services;
    private final OnServiceClickListener listener;

    public ServiceAdapter(List<RepairService> services, OnServiceClickListener listener) {
        this.services = services;
        this.listener = listener;
    }

    public static class ServiceViewHolder extends RecyclerView.ViewHolder {
        TextView tvName, tvDescription, tvPrice;

        public ServiceViewHolder(@NonNull View itemView) {
            super(itemView);
            tvName = itemView.findViewById(R.id.tvServiceName);
            tvDescription = itemView.findViewById(R.id.tvServiceDescription);
            tvPrice = itemView.findViewById(R.id.tvServicePrice);
        }
    }

    @NonNull
    @Override
    public ServiceViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_service, parent, false);
        return new ServiceViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ServiceViewHolder holder, int position) {
        RepairService service = services.get(position);
        holder.tvName.setText(service.name);
        holder.tvDescription.setText(service.description);
        holder.tvPrice.setText(String.format(Locale.getDefault(), "Rs. %.2f", service.price));
        holder.itemView.setOnClickListener(v -> listener.onServiceClick(service));
    }

    @Override
    public int getItemCount() {
        return services.size();
    }
}

File: app/src/main/java/com/example/techfix/ui/ServiceAdapter.kt
package com.example.techfix.ui

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.techfix.R
import com.example.techfix.data.entity.RepairService

class ServiceAdapter(
    private val services: List<RepairService>,
    private val onServiceClick: (RepairService) -> Unit
) : RecyclerView.Adapter<ServiceAdapter.ServiceViewHolder>() {

    class ServiceViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val tvName: TextView = view.findViewById(R.id.tvServiceName)
        val tvDescription: TextView = view.findViewById(R.id.tvServiceDescription)
        val tvPrice: TextView = view.findViewById(R.id.tvServicePrice)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ServiceViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_service, parent, false)
        return ServiceViewHolder(view)
    }

    override fun onBindViewHolder(holder: ServiceViewHolder, position: Int) {
        val service = services[position]
        holder.tvName.text = service.name
        holder.tvDescription.text = service.description
        holder.tvPrice.text = "Rs. %.2f".format(service.price)
        holder.itemView.setOnClickListener { onServiceClick(service) }
    }

    override fun getItemCount(): Int = services.size
}